packages/types/config/runtime.d.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * NuxtConfigurationRuntimeConfig
+ * Documentation : TBD
+ * NuxtRuntimeConfig interface can be extended by users to enable intellisense on $config
+ */
+
+type JSONObject = { [key: string]: JSONValue }
+type JSONValue = null | undefined | boolean | string | number | JSONObject | JSONValue[]
+
+export interface NuxtRuntimeConfig extends JSONObject {}
+
+export type NuxtConfigurationRuntimeConfig = NuxtRuntimeConfig | ((env: typeof process.env) => NuxtRuntimeConfig)
```

Comment:
I'm still about uncertain about those since same types will be used both before and after "normalization". So if user extends the types, he might run into some problems with types being different for the same property. For example user extends:

```ts
interface NuxtRuntimeConfig {
  foo: boolean
}
```

then:

```
privateRuntimeConfig: {
  foo: process.env.FOO  // FOO === 'true'
}
```

And at that point the `foo` type will be wrong. Only after normalization, it will actually be converted to bool.

Reply:
But I don't know of any possible way to handle that properly.

Reply:
Unfortunately, we can't normalize for `process.env` (it is a big breaking change for using dotenv as by default all vals are string) but one can simply do:

```js
privateRuntimeConfig: {
  FOO: false
}
```

If inside `.env` or `process.env` we set `FOO: "true"`, `$config.FOO` is `true` (not `"true"`)

-- Alternatively one can do:

```js
privateRuntimeConfig: {
  foo: process.env.FOO === 'true'
}
```

or:

```js
privateRuntimeConfig: {
  foo: destr(process.env.FOO)
}
```

or:

```js
privateRuntimeConfig: env => ({ foo: env.FOO })
```